reo/src/reopt.jl

```diff
@@ -644,16 +644,48 @@ function add_tou_demand_charge_constraints(m, p)
 		sum( m[:dvGridPurchase][u, ts] for u in p.PricingTier )
 	)
 	
-	##Constraint (12e): Peak demand used in percent lookback calculation 
-	@constraint(m, [m in p.DemandLookbackMonths],
-		m[:dvPeakDemandELookback] >= sum(m[:dvPeakDemandEMonth][m, n] for n in p.DemandMonthsBin)
-	)
-	
-	##Constraint (12f): Ratchet peak demand charge is bounded below by lookback
-	@constraint(m, [r in p.DemandLookbackMonths],
-		sum( m[:dvPeakDemandEMonth][r,e] for e in p.DemandBin ) >= 
-		p.DemandLookbackPercent * m[:dvPeakDemandELookback] 
-	)
+	if p.DemandLookbackRange != 0  # then the dvPeakDemandELookback varies by month
+		
+		##Constraint (12e): dvPeakDemandELookback is the highest peak demand in DemandLookbackMonths
+		for mth in p.Month
+			if p.Month > p.DemandLookbackRange
+				@constraint(m, [lm in 1:p.DemandLookbackRange],
+					m[:dvPeakDemandELookback][mth] >= 
+					sum(m[:dvPeakDemandEMonth][mth - lm, n] for n in p.DemandMonthsBin)
+				)
+			else  # need to handle rollover months
+				for lm in 1:p.DemandLookbackRange
+					lkbkmonth = mth - lm
+					if lkbkmonth ≤ 0
+						lkbkmonth += 12
+					end
+					@constraint(m,
+						m[:dvPeakDemandELookback][mth] >= 
+						sum(m[:dvPeakDemandEMonth][lkbkmonth, n] for n in p.DemandMonthsBin)
+					)
+				end
```

Comment:
Would the logic under "else" work for all months, to avoid the if-else? (If breaking this into if and else speeds up code, please ignore this comment!)

Reply:
the if-else is to handle lookback months into the "previous" year (eg. month = 1, lookback = 3), I couldn't think of way to handle it in one block

Reply:
@kkrah and it might speed things up having the vectorized constraints in the `if` block.